sccp/figures/figureThomsonDrop.py

```diff
@@ -1,41 +1,202 @@
+    )
+    plotDifferentialExpression(data, "CTRL4", "NK Cells", NKCellGeneSet, rank, *ax[8:12])
+    plotDifferentialExpression(
+        data, "CTRL4", "DCs", DCCellGeneSet, rank, *ax[12:16]
+    )
+    plotDifferentialExpression(
+        data, "CTRL4", "pDCs", pDCGeneSet, rank, *ax[16:20], ct2=True,
+    )
+    plotDifferentialExpression(data, "CTRL4", "T Cells", TCellGeneSet, rank, *ax[20:24])
+    plotDifferentialExpression(
+        data, "CTRL4", "Cytotoxic T Cells", CytoTCellGeneSet, rank, *ax[24:28], ct2=True
+    )
+    plotDifferentialExpression(
+        data, "CTRL4", "Memory T Cells", memorytcellgeneset, rank, *ax[28:32], ct2=True
+    )
+    plotDifferentialExpression(
+        data, "CTRL4", "cDCs", cDCgeneset, rank, *ax[32:36], ct2=True
+    )
+    return f
 
-    origX = pf2(data, rank, doEmbedding=False)
 
-    plotConditionsFactors(
-        origX, ax[0], groupDrugs(origX.obs["Condition"]), ThomsonNorm=True
-    )
-    plotCellState(origX, ax[1])
-    plotGeneFactors(origX, ax[2])
+def plotDifferentialExpression(
+    data,
+    condition,
+    cell_type,
+    geneset,
+    rank,
+    *args,
+    ct2=False,
+    override=None
+):
+    sampled_data = None
+    if not ct2:
+        sampled_data = data[
+            (data.obs["Cell Type"] != cell_type) | (data.obs["Condition"] != condition)
+        ]
+    else:
+        sampled_data = data[
+            (data.obs["Cell Type2"] != cell_type) | (data.obs["Condition"] != condition)
+        ]
 
-    # sampled_data = data
+    origX = pf2(data, rank, doEmbedding=False)
 
     sampledX = pf2(sampled_data, rank, doEmbedding=False)
 
-    plotConditionsFactors(
-        sampledX, ax[3], groupDrugs(origX.obs["Condition"]), ThomsonNorm=True
-    )
-    plotCellState(sampledX, ax[4])
-    plotGeneFactors(sampledX, ax[5])
+    isolated_orig_vals = plotGeneFactorsIsolated(origX, args[0], geneset)
+    isolated_vals = plotGeneFactorsIsolated(sampledX, args[1], geneset)
+    most_exp_cmp = -1
+    most_exp = -1
+    most_exp_cmp2 = -1
+    most_exp2 = -1
 
-    return f
+    if override is None:
+        for i in range(isolated_orig_vals.shape[1]):
+            exp_sum = np.sum(np.abs(isolated_orig_vals[:, i]))
+            if exp_sum > most_exp:
+                most_exp = exp_sum
+                most_exp_cmp = i
+        for i in range(isolated_vals.shape[1]):
+            exp_sum = np.sum(np.abs(isolated_vals[:, i]))
+            if exp_sum > most_exp2:
+                most_exp2 = exp_sum
+                most_exp_cmp2 = i
+    else:
+        most_exp_cmp = override[0]
+        most_exp_cmp2 = override[1]
+    X = np.array(origX.uns["Pf2_A"])
+    X = X[:, most_exp_cmp]
+    Y = np.array(sampledX.uns["Pf2_A"])
+    Y = Y[:, most_exp_cmp2]
+
+    yt = pd.Series(np.unique(origX.obs["Condition"]))
+    yt2 = pd.Series(np.unique(sampledX.obs["Condition"]))
+
+    assert yt.equals(yt2)
+    if not ct2:
+        print(f"Number of {cell_type}: {len(data[data.obs['Cell Type'] == cell_type])}")
```

Comment:
Almost all of this is the same. Splice out the difference to simplify the if statement.